xlsynth-driver/src/dslx_equiv.rs

```diff
@@ -25,11 +27,16 @@ fn build_ir_for_dslx(
     disable_warnings: Option<&[String]>,
     tool_path: Option<&str>,
     type_inference_v2: Option<bool>,
+    want_domains: bool,
```

Comment:
should we call this want_enum_domains or do you forsee more things beyond enums? if so might be good to put a function level comment on the param's generality